File: Leetcode 229. Majority Element II/lc 229.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> res=new ArrayList<>();
        int len=nums.length;
        int min=len/3;
        HashMap<Integer,Integer> hm=new HashMap<>();
        for(int i=0;i<len;i++){
            hm.put(nums[i],hm.getOrDefault(nums[i],0)+1);
        }

        for(Map.Entry<Integer,Integer> m:hm.entrySet()){
            int value=m.getValue();
            if(value>min){
                res.add(m.getKey());
            }
        }

        return res;
    }
}

File: Leetcode 229. Majority Element II/lc 229.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> ans;
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        int min=nums.size()/3;

        for(auto i:mp){
            if(i.second>min){
                ans.emplace_back(i.first);
            }
        }
        return ans;
    }
};